pkg/sinks/sink.go

```diff
@@ -111,16 +111,7 @@ func (u *SinkProcessor) Start(ctx context.Context) error {
 			return
 		}
 	}()
-	var metricsOpts []metrics.Option
-	if udSink := u.VertexInstance.Vertex.Spec.Sink.UDSink; udSink != nil {
-		if serverHandler, ok := sinker.(*udsink.UserDefinedSink); ok {
-			metricsOpts = metrics.NewMetricsOptions(ctx, u.VertexInstance.Vertex, serverHandler, reader, nil)
-		} else {
-			return fmt.Errorf("unable to get the metrics options for the udsink")
-		}
-	} else {
-		metricsOpts = metrics.NewMetricsOptions(ctx, u.VertexInstance.Vertex, nil, reader, nil)
-	}
+	metricsOpts := metrics.NewMetricsOptions(ctx, u.VertexInstance.Vertex, nil, reader, nil)
```

Comment:
should we change the signature of `NewMetricsOptions` since we are passing `nil` for the checker.

Reply:
Technically, that still could be used for other kinds of health check related to the platform if needed, let's keep it there for now.